mmedit/models/restorers/basicvsr.py

```diff
@@ -0,0 +1,182 @@
+import numbers
+import os.path as osp
+
+import mmcv
+import numpy as np
+import torch
+
+from mmedit.core import tensor2img
+from ..registry import MODELS
+from .basic_restorer import BasicRestorer
+
+
+@MODELS.register_module()
+class BasicVSR(BasicRestorer):
+    """BasicVSR model for video super-resolution.
+
+    Note that this model is used for IconVSR.
+
+    Paper:
+        BasicVSR: The Search for Essential Components in Video Super-Resolution
+        and Beyond, CVPR, 2021
+
+    Args:
+        generator (dict): Config for the generator structure.
+        pixel_loss (dict): Config for pixel-wise loss.
+        train_cfg (dict): Config for training. Default: None.
+        test_cfg (dict): Config for testing. Default: None.
+        pretrained (str): Path for pretrained model. Default: None.
+    """
+
+    def __init__(self,
+                 generator,
+                 pixel_loss,
+                 train_cfg=None,
+                 test_cfg=None,
+                 pretrained=None):
+        super(BasicVSR, self).__init__(generator, pixel_loss, train_cfg,
+                                       test_cfg, pretrained)
+
+        # fix pre-trained networks
+        self.fix_iter = train_cfg.get('fix_iter', 0) if train_cfg else 0
+        self.generator.find_unused_parameters = False
+
+        # count training steps
+        self.step_counter = train_cfg.get('step_counter',
```

Comment:
Proposal:

Add a  `TAIL` component to the module, just like backbones and heads, and save step counters there, so that they could be automatically saved and restored.